pyproject.toml

```diff
@@ -65,6 +65,7 @@ flake8-black = "^0.2.1"
 django = {version = ">=2,<4", optional = false}
 pydantic = {version = "^1.6.1", optional = false}
 uvicorn = ">=0.11.6,<0.13.0"
+promise = "^2.3"
```

Comment:
should we make this optional? 🤔 

also I found this the other day, wonder if it is a good alternative https://github.com/celery/vine (maybe they are swappable)

Reply:
Ooh vine looks good, I'll look into that. The promise library is not maintained so definitely in favour so switching to something else.